Modules/Core/Mesh/test/itkConnectedRegionsMeshFilterTest2.cxx

```diff
@@ -0,0 +1,82 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include "itkImageFileReader.h"
+#include "itkConnectedRegionsMeshFilter.h"
+#include "itkTestingMacros.h"
+#include "itkBinaryThresholdImageFilter.h"
+#include "itkBinaryMask3DMeshSource.h"
+#include <iostream>
+
+/*
+ * Test the mesh connectivity class.
+ */
+int
+itkConnectedRegionsMeshFilterTest2(int, char * argv[])
+{
+
+  // Read 3D Image to create Test Mesh
+  using PixelType = unsigned char;
+  const unsigned int Dimension = 3;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  using ReaderType = itk::ImageFileReader<ImageType>;
+  auto reader = ReaderType::New();
+  reader->SetFileName(argv[1]);
+  reader->Update();
+
+  auto inputImage = reader->GetOutput();
+
+  // Threshold the 3D image to get binary mask for mesh generation
+  using ThresholdFilterType = itk::BinaryThresholdImageFilter<ImageType, ImageType>;
+  auto thresholdFilter = ThresholdFilterType::New();
+  thresholdFilter->SetInput(inputImage);
+  thresholdFilter->SetLowerThreshold(0);
+  thresholdFilter->SetUpperThreshold(200);
+  thresholdFilter->SetOutsideValue(1);
+  thresholdFilter->SetInsideValue(0);
+  thresholdFilter->Update();
+
+  auto outputImage = thresholdFilter->GetOutput();
+
+  // Get mesh from binary image
+  using MeshType = itk::Mesh<float, 3>;
+  using MeshSourceType = itk::BinaryMask3DMeshSource<ImageType, MeshType>;
+
+  auto meshSource = MeshSourceType::New();
+  meshSource->SetInput(outputImage);
+  meshSource->SetObjectValue(1);
+  meshSource->Update();
+  auto mesh = meshSource->GetOutput();
+
+  // Check number of connected components in the mesh
+  using ConnectFilterType = itk::ConnectedRegionsMeshFilter<MeshType, MeshType>;
+  auto connectivityFilter = ConnectFilterType::New();
+  connectivityFilter->SetInput(mesh);
+  connectivityFilter->SetExtractionModeToAllRegions();
+  connectivityFilter->Update();
+
+  ITK_TEST_EXPECT_TRUE(connectivityFilter->GetNumberOfExtractedRegions() == 15);
```

Comment:
Eventually, for another input mesh (e.g. if the filter is found not to be working for that particular mesh), we'd like to be able to check this dynamically, according to the value of an input parameter that is provided together with the image.

Reply:
In the current test implementation, I am generating a mesh from a 3D image file. 
So its future application to test directly on other mesh is not present. 

What I can do is to check if the input file is not a .vtk then continue with the current approach and if the input file is a .vtk then directly work on that as a mesh source. Also I can make the number of cells, and connected components as input parameters to the test.

Reply:
Even in the case where the input was another image file, the previous implementation was not allowing to test it. Thanks for considering mesh files as well.